src/frequenz/sdk/actor/power_distributing/_component_managers/_ev_charger_manager/_states.py

```diff
@@ -0,0 +1,132 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Power distribution state tracking for ev chargers."""
+
+
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Iterable
+
+from frequenz.client.microgrid import EVChargerData
+
+from .....timeseries import Power
+
+
+@dataclass
+class EvcState:
+    """A class for tracking state of an ev charger."""
+
+    component_id: int
+    """The component id of the ev charger."""
+
+    last_data: EVChargerData
+    """The last data received from the EV charger."""
+
+    power: Power
+    """The power currently used by the EV charger."""
+
+    last_allocation: Power
+    """The last allocation made for the EV."""
+
+    last_reallocation_time: datetime
+    """The last time the ev charger was allocated power.
+
+    Used to make sure we don't allocate power to the ev charger too often.
+    """
+
+    last_charging_time: datetime
+    """The last time the ev charger was charging.
+
+    Used to de-allocate power from the ev charger if it has not been charging
+    for a while.
+    """
+
+    def update_last_allocation(self, allocation: Power, alloc_time: datetime) -> None:
+        """Update the last allocation and related timestamps.
+
+        Args:
+            allocation: The most allocation allocation made for the EV.
+            alloc_time: The time at which the allocation was made.
+        """
+        self.last_allocation = allocation
+        self.last_reallocation_time = alloc_time
+        self.last_charging_time = alloc_time
+
+    def update_state(
+        self,
+        latest_ev_data: EVChargerData,
+    ) -> None:
+        """Update EvcState from component data.
+
+        Args:
+            latest_ev_data: latest ev data from component data stream.
+        """
+        self.power = Power.from_watts(latest_ev_data.active_power)
+        self.last_data = latest_ev_data
+
+        if self.power > Power.zero():
+            self.last_charging_time = latest_ev_data.timestamp
+
+
+class EvcStates:
+    """Tracks states of all ev chargers."""
+
+    _states: dict[int, EvcState]
+
+    def __init__(self) -> None:
+        """Initialize the EvcStates object."""
```

Comment:
Also super minor and optional: I started using just "Initialize this instance." so it is more copy&pasteable and more renaming friendly.

```suggestion
        """Initialize this instance."""
```